fahrschein/src/main/java/org/zalando/fahrschein/NakadiClient.java

```diff
@@ -95,8 +109,13 @@ public <T> void publish(String eventName, List<T> events) throws EventPublishing
             objectMapper.writeValue(body, events);
         }
 
+        requestHandlers.forEach(requestHandler -> requestHandler.beforeExecute(request));
```

Comment:
I would suggest writing a test for the execution of the handlers.

Reply:
Plus one for a test. It should cover the behaviour if `beforeExecute` etc. throws.